ic-stable-structures/src/structure/stable_storage/btreemap.rs

```diff
@@ -1,39 +1,38 @@
 use std::ops::RangeBounds;
 
-use dfinity_stable_structures::memory_manager::MemoryId;
-use dfinity_stable_structures::{btreemap, BoundedStorable};
+use dfinity_stable_structures::{btreemap, BoundedStorable, Memory};
 
-use super::get_memory_by_id;
 use crate::structure::BTreeMapStructure;
-use crate::{IterableSortedMapStructure, Memory};
+use crate::IterableSortedMapStructure;
 
 /// Stores key-value data in stable memory.
-pub struct StableBTreeMap<K, V>(btreemap::BTreeMap<K, V, Memory>)
+pub struct StableBTreeMap<K, V, M: Memory>(btreemap::BTreeMap<K, V, M>)
 where
     K: BoundedStorable + Ord + Clone,
     V: BoundedStorable;
 
-impl<K, V> StableBTreeMap<K, V>
+impl<K, V, M> StableBTreeMap<K, V, M>
 where
     K: BoundedStorable + Ord + Clone,
     V: BoundedStorable,
+    M: Memory,
 {
     /// Create new instance of key-value storage.
-    pub fn new(memory_id: MemoryId) -> Self {
-        let memory = get_memory_by_id(memory_id);
+    pub fn new(memory: M) -> Self {
```

Comment:
This change in the `new` fn (and in all of the stable_storage structures) makes this contructor different than the one in the `heap` structures. This breaks the `always-heap` feature

Reply:
BTW, do we need the `heap` structures? Can we maybe use structures with vector memory instead?

Reply:
I've made those containers compatible again